src/xma/src/xmaplugin/xmaplugin.cpp

```diff
@@ -710,43 +671,12 @@ XmaCUCmdObj xma_plg_schedule_cu_cmd(XmaSession s_handle,
         itr++;
     }
 
-    // Setup ert_start_kernel_cmd 
-    ert_start_kernel_cmd *cu_cmd = 
-        (ert_start_kernel_cmd*)priv1->kernel_execbos[bo_idx].data;
-    cu_cmd->state = ERT_CMD_STATE_NEW;
-    if (kernel_tmp1->soft_kernel) {
-        cu_cmd->opcode = ERT_SK_START;
-    } else {
-        cu_cmd->opcode = ERT_START_CU;
-    }
-    cu_cmd->extra_cu_masks = 3;//XMA now supports 128 CUs
-
-    cu_cmd->cu_mask = kernel_tmp1->cu_mask0;
-
-    cu_cmd->data[0] = kernel_tmp1->cu_mask1;
-    cu_cmd->data[1] = kernel_tmp1->cu_mask2;
-    cu_cmd->data[2] = kernel_tmp1->cu_mask3;
+    priv1->kernel_execbos[bo_idx].xrt_run = xrt::run(priv1->kernel_execbos[bo_idx].xrt_kernel);
+    ert_start_kernel_cmd* cu_cmd =
+        reinterpret_cast<ert_start_kernel_cmd*>(priv1->kernel_execbos[bo_idx].xrt_run.get_ert_packet());
```

Comment:
Do not repeat the type
```suggestion
    auto cu_cmd = static_cast<ert_start_kernel_cmd*>(priv1->kernel_execbos[bo_idx].xrt_run.get_ert_packet());
```

Reply:
seeing compilation error if we use static_cast here.  using auto now, not repeating the type. submitted latest changes in the PR.